codegen/tests/simple/app.go

```diff
@@ -0,0 +1,89 @@
+// Code generated by sysl DO NOT EDIT.
+package simple
+
+import (
+	"context"
+
+	"github.com/anz-bank/sysl-go/codegen/tests/deps"
+	"github.com/anz-bank/sysl-go/config"
+	"github.com/anz-bank/sysl-go/core"
+	"github.com/anz-bank/sysl-go/handlerinitialiser"
+)
+
+// HandlerManager for Simple
+type HandlerManager struct {
+	coreCfg             *config.DefaultConfig
+	enabledHandlers     []handlerinitialiser.HandlerInitialiser
+	enabledGrpcHandlers []handlerinitialiser.GrpcHandlerInitialiser
+}
+
+// NewHandlerManager create new HandlerManager for Simple
+func NewHandlerManager(coreCfg *config.DefaultConfig) *HandlerManager {
+	mgr := &HandlerManager{
+		coreCfg:             coreCfg,
+		enabledHandlers:     []handlerinitialiser.HandlerInitialiser{},
+		enabledGrpcHandlers: nil,
+	}
+
+	return mgr
+}
+
+// EnabledHandlers ...
+func (h *HandlerManager) EnabledHandlers() []handlerinitialiser.HandlerInitialiser {
+	return h.enabledHandlers
+}
+
+// EnabledGrpcHandlers ...
+func (h *HandlerManager) EnabledGrpcHandlers() []handlerinitialiser.GrpcHandlerInitialiser {
+	return h.enabledGrpcHandlers
+}
+
+// LibraryConfig ...
+func (h *HandlerManager) LibraryConfig() *config.LibraryConfig {
+	return &h.coreCfg.Library
+}
+
+// LoadServices ...
+func (h *HandlerManager) LoadServices(ctx context.Context) error {
+	bffSI := ServiceInterface{
+		GetApiDocsList:            simpleGetApiDocsList,
+		GetJustOkAndJustErrorList: simpleGetJustOkAndJustErrorList,
+		GetJustReturnErrorList:    simpleGetJustReturnErrorList,
+		GetJustReturnOkList:       simpleGetJustReturnOkList,
+		GetOkTypeAndJustErrorList: simpleGetOkTypeAndJustErrorList,
+		GetOopsList:               simpleGetOopsList,
+		GetRawList:                simpleGetRawList,
+		GetRawIntList:             simpleGetRawIntList,
+		GetStuffList:              simpleGetStuffList,
+		PostStuff:                 simplePostStuff,
+	}
+
+	genCallbacks := config.Callback{
+		UpstreamTimeout:   h.coreCfg.GenCode.Upstream.ContextTimeout,
+		DownstreamTimeout: h.coreCfg.GenCode.Downstream.(*DownstreamConfig).ContextTimeout,
+		ErrorHandler:      config.ErrorHandler{},
+		RouterBasePath:    "/",
+		UpstreamConfig:    h.coreCfg.GenCode.Upstream,
+	}
+
+	depsHTTPClient, depsErr := core.BuildDownstreamHTTPClient("deps", &h.coreCfg.GenCode.Downstream.(*DownstreamConfig).Deps)
```

Comment:
To keep changes small, the section below (to the end of the function) should be split out into a separate method and this should be the focus of the first pull request. Generated code should be something like:

```go
func InitialiseHandlers(handlerManager *HandlerManager, serviceInterface ServiceInterface, callbacks GenCallbacks) error {
    // build the client
    // build the service handler
    // build the service router
    // enable the handlers
}
```


The service interface will always be custom code, never generated. The construction of the callbacks will also be custom at this stage. So custom code can look something like:

```go
func (h *HandlerManager) LoadServices(ctx context.Context) error {
    serviceInterface := ...
    serviceCallback := ...
    error := InitialiseHandler(h, serviceInterface, serviceCallback)
    ...
}
```

Reply:
Removing handler manager from the code and just generating the above portion would break the build, how do we plan to handle that?

Reply:
Included the HandlerManager struct in the app, so that build will go through.

Reply:
The `App` isn't doing much here. Can we change this to a function instead:
```
type HandlerInit struct {
	HttpHandlers []handlerinitialiser.HandlerInitialiser
	GrpcHandlers []handlerinitialiser.GrpcHandlerInitialiser
}

func InitialiseHandlers(coreCfg *config.DefaultConfig, serviceInterface *ServiceInterface, serviceCallback core.RestGenCallback) (*HandlerInit, error) {
    ...
    serviceRouter := ...
    httpHandlers := []handlerinitialiser.HandlerInitialiser{serviceRouter}
    return &HandlerInit{HttpHandlers: httpHandlers}, nil
}
```

Reply:
The intent is have App is to have content into it, in the future.

Reply:
Then let's make that change in the future when we know what it should look like.

Reply:
I can add this to handlerManger, that should be easy.

Reply:
As a name, `HandlerManager` is ambiguous. This function doesn't need to operate on an object because the only purpose of the object would be to invoke its single method. 

Make it a function for now. `InitialiseHandlers` returns the initialised handlers.

Reply:
ok